src/modalities/dataloader/dataset.py

```diff
@@ -254,40 +253,60 @@ def __len__(self) -> int:
         """
         return len(self._index)
 
-    def __getitem__(self, idx: int) -> BatchEncoding:
+    def __getitem__(self, idx: int | slice) -> BatchEncoding:
         """
-        Retrieves the item at the given index.
+        Retrieves the item at the given index or a slice of items.
 
         Args:
-            idx (int): The index of the item to retrieve.
+            idx (int | sclice): The index of the item to retrieve or slice of items.
 
         Returns:
             BatchEncoding: The retrieved item as a BatchEncoding object.
 
         Raises:
             ValueError: If the length of the sample in bytes is not a multiple of `self._token_size_in_bytes`.
         """
-        self._check_if_inbounds(idx)
-        # offset and length in bytes
-        offset_in_bytes, length_in_bytes = self._index[idx]
-        if length_in_bytes % self._token_size_in_bytes != 0:
-            raise ValueError(
-                f"Length of the sample in bytes is not a multiple of {self._token_size_in_bytes}."
-                f"Offset in bytes: {offset_in_bytes}, Length in bytes: {length_in_bytes}"
-            )
+
+        if not isinstance(idx, slice):
+            # (offset_in_bytes, length_in_bytes)
+            item_positions: list[tuple[int, int]] = [self._index[idx]]
+        else:
+            if idx.step is not None and idx.step != 1:
+                raise ValueError("Slicing with step != 1 is not supported.")
+            item_positions = self._index[idx]
+
+        if len(item_positions) == 0:
```

Comment:
Is it safe to not throw an exception in this case?

Reply:
This is the convention in slicing. 
Examples are: 

```py
[0,1,2,3,4,5]][4:2] == []
[0,1,2,3,4,5]][6:] == []
```

Thus, if we receive an empty list from `self._index[idx]`, we can just return an empty `BatchEncoding `.

Reply:
The dataloader uses the `PackedMemMapDatasetBase.__get_item__()` for retrieving the samples. 

However, the dataloader does not slice, as it uses direct indexing instead. 

If the index is out of bounds an error is being thrown. 

```py
>>> [0,1,2,3,4][10:]    # slicing   
[]
>>> [0,1,2,3,4][10]    # indexing 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```